### DISCUSSION TITLE
Welcome to plugin-installation-manager-tool Discussions!

### DISCUSSION THREAD

**Discussion by timja** [jenkinsci/plugin-installation-manager-tool#244] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by hp77-creator** [comment]
Hello @timja , I am Himanshu Pandey, an undergrad, looking to contribute to the jenkins organisation and take part in the GSoC. There is a project idea this year on this tool. And in that idea, Oleg has mentioned, student must try to integrate it with docker and configuration as code. Afaik and also reiterated same by mark in gsoc sig meetings that it is already integrated, so is it integrated with docker or what we are seeing is a work around and there is still work to be done? I don't know much about CasC but I would like to know your view and opinion on how one must look for the integration of tool in CasC? Thanks.

**Reply by timja** [reply]
so it's already integrated in the docker image https://github.com/jenkinsci/docker#plugin-installation-manager-cli-preview

I vaguely remember in regards to configuration as code that it ignores unknown root elements so you can share the `jenkins.yaml` file between plugin manager and configuration as code

**Comment by oleg-nenashev** [comment]
Yes, the idea description is outdated. Sorry about that, I have not got to
it really :(

**Comment by sirkkalap** [comment]
Hi wonderful people,

I read the README.md and it was still a bit unclear to me why: "The goal of this tool is to replace the Docker install-plugins.sh script and the many other implementations of plugin management that have been recreated across Jenkins."?

Is it that other implementations lack something or is the reason for another tool something deeper than that?

Why would I choose one or the other?

Yours: Petri Sirkkala

**Reply by timja** [reply]
the install-plugins.sh script was a very hacky shell script that couldn't be safely extended

this was written to replace it basically...

**Comment by ricksbrown** [comment]
### The Rather Peculiar `--output` arg.

I must say, the `--output` arg seems rather confused:  `--output {stdout,yaml,txt}`

Seems like mixed categories there, since one of the options is `stdout` I would expect the other options to _not_ be stdout.
In fact all three options go to stdout.

So either `stdout` is not named correctly OR `yaml` and `txt` should actually go somewhere other than stdout (pehaps a yaml file or a text file respectively).

**Reply by timja** [reply]
I think it's named poorly.

It's more like 'human', but that probably isn't a good name.

From what I remember it's human readable text rather than a 'machine' file

**Comment by sai-praveenkumar** [comment]
Hello Developers and Mentors,I am a final year student and I would like to contribute to this project,but I don't know a single thing about where to get started and I'm stuck.Could you please help me through what to learn and contribute to this project please.

**Reply by timja** [reply]
Hi there's a gitter room here:
https://gitter.im/jenkinsci/plugin-installation-manager-cli-tool

Some contributing guides on jenkins.io:
https://www.jenkins.io/participate/code/

and issues here:
https://github.com/jenkinsci/plugin-installation-manager-tool/issues
